plugins/net_plugin/net_plugin.cpp

```diff
@@ -3180,6 +3254,22 @@ namespace eosio {
       return true;
    }
 
+   // called from connection strand
+   void connection::send_block_nack(const block_id_type& block_id) {
+      if (protocol_version < proto_block_nack || my_impl->p2p_disable_block_nack)
+         return;
+
+      if (my_impl->sync_master->syncing_from_peer())
+         return;
+
+      peer_dlog(this, "Sending nack ${n}", ("n", block_header::num_from_id(block_id)));
+
+      buffer_factory buff_factory;
+      auto send_buffer = buff_factory.get_send_buffer( block_nack_message{block_id} );
+
+      enqueue_buffer( msg_type_t::block_nack_message, false, send_buffer, 0, no_reason );
```

Comment:
I feel that these short messages should use the sync queue.

Reply:
The block sync queue is specifically for blocks that were requested. It exists solely so that processed blocks are sent out after requested blocks.